ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/forest/worldview/ForestMutableWorldState.java

```diff
@@ -63,18 +64,22 @@ public class ForestMutableWorldState implements MutableWorldState {
   private final Map<Bytes32, Address> newAccountKeyPreimages = new HashMap<>();
 
   public ForestMutableWorldState(
-      final WorldStateStorage storage,
+      final WorldStateKeyValueStorage worldStateKeyValueStorage,
       final WorldStatePreimageStorage preimageStorage,
       final EvmConfiguration evmConfiguration) {
-    this(MerkleTrie.EMPTY_TRIE_NODE_HASH, storage, preimageStorage, evmConfiguration);
+    this(
+        MerkleTrie.EMPTY_TRIE_NODE_HASH,
+        worldStateKeyValueStorage,
+        preimageStorage,
+        evmConfiguration);
   }
 
   public ForestMutableWorldState(
       final Bytes32 rootHash,
-      final WorldStateStorage worldStateStorage,
+      final WorldStateKeyValueStorage worldStateKeyValueStorage,
       final WorldStatePreimageStorage preimageStorage,
       final EvmConfiguration evmConfiguration) {
-    this.worldStateStorage = worldStateStorage;
+    this.worldStateKeyValueStorage = (ForestWorldStateKeyValueStorage) worldStateKeyValueStorage;
```

Comment:
we can probably get away from this explicit cast by changing the constructor signature.   It looks like only evmtool BlockchainModule is calling this from a context that doesn't explicitly have storage as a  ForestWorldStateKeyValueStorage.